### PR TITLE
Docs: Added Solutions to Leetcode 621

### LINKED ISSUE #1379: [Feature Request]: Leetcode Problem 621 - (Task Scheduler-Medium)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Provide Solution To Leetcode Problem 621 - (Task Scheduler)

### Use Case

I would add proper solutions for the problem in three different programing languages i.e C++,Java,and Python with proper documentation

### Benefits

Users can access solution to problem 621

### Add ScreenShots

N/A

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by shreyash3087**
## Related Issue

Closes #1379

## Description

This PR adds the solution to leetcode problem 6621 - (Task Scheduler)

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [X] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

screencapture-localhost-3000-dsa-solutions-lc-solutions-0600-0699-task-scheduler-2024-06-16-18_49_38

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have followed the code style guidelines of this project.
- [X] I have checked for any existing open issues that my pull request may address.
- [X] I have ensured that my changes do not break any existing functionality.
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X] I have read the resources for guidance listed below.
- [X] I have followed security best practices in my code changes.

## Additional Context

N/A

### CODE REVIEW

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+Output: 6
+
+Explanation: A possible sequence is: A -> B -> C -> D -> A -> B.
+
+With a cooling interval of 1, you can repeat a task after just one other task.
+```
+
+### Constraints
+
+- $1 <= tasks.length <= 10^4$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:46]: replace `$1 <= tasks.length <= 10^4$` to `$1 \leq \text{tasks.length} \leq 10^4$`

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+- `tasks[i]` is an uppercase English letter.
+- $0 <= n <= 100$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:48]: replace `$0 <= n <= 100$` to `$0 \leq n \leq 100$`

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+## Complexity Analysis
+
+### Time Complexity: $O(N)$
+
+> **Reason**:  In the worst case, all tasks must be processed, and each task might be inserted and extracted from the priority queue. The priority queue operations (insertion and extraction) have a time complexity of O(log⁡k) each. Therefore, the overall time complexity is O(N⋅log⁡k). Since k is at maximum 26, log⁡k is a constant term. We can simplify the time complexity to O(N). This is a linear time complexity with a high constant factor..
+
+### Space Complexity: $O(1)$
+
+> **Reason**: The space complexity is mainly determined by the frequency array and the priority queue. The frequency array has a constant size of 26, and the priority queue can have a maximum size of 26 when all distinct tasks are present. Therefore, the overall space complexity is O(1) or O(26), which is considered constant.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:271]: replace `O(1)` to `$O(1)$`, `O(26)` to `$O(26)$`

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+
+### Approach #2 Filling the Slots and Sorting
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:273]: replace `#2` to `2`

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+
+### Space Complexity: $O(1)$
+
+> **Reason**: The frequency array has a size of 26.
+>
+> Note that some extra space is used when we sort arrays in place. The space complexity of the sorting algorithm depends on the programming language.
+>
+>   - In Python, the sort method sorts a list using the Timsort algorithm which is a combination of Merge Sort and Insertion Sort and has O(N) additional space.
+>   - In Java, Arrays.sort() is implemented using a variant of the Quick Sort algorithm which has a space complexity of O(log⁡N) for sorting two arrays.
+>   - In C++, the sort() function is implemented as a hybrid of Quick Sort, Heap Sort, and Insertion Sort, with a worse-case space complexity of O(log⁡N).
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:411]: replace `v` to `$O(log⁡N)$`

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+
+> **Reason**:  The time complexity of the algorithm is O(26log⁡26+N), where 26log⁡26 is the time complexity of sorting the frequency array, and N is the length of the input task list, which is the dominating term.
+
+### Space Complexity: $O(1)$
+
+> **Reason**: The frequency array has a size of 26.
+>
+> Note that some extra space is used when we sort arrays in place. The space complexity of the sorting algorithm depends on the programming language.
+>
+>   - In Python, the sort method sorts a list using the Timsort algorithm which is a combination of Merge Sort and Insertion Sort and has O(N) additional space.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:409]: replace `O(N)` to `$O(N)$`

File: dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+
+### Time Complexity: $O(N)$
+
+> **Reason**:  The time complexity of the algorithm is O(26log⁡26+N), where 26log⁡26 is the time complexity of sorting the frequency array, and N is the length of the input task list, which is the dominating term.
+
+### Space Complexity: $O(1)$
+
+> **Reason**: The frequency array has a size of 26.
+>
+> Note that some extra space is used when we sort arrays in place. The space complexity of the sorting algorithm depends on the programming language.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:407]: If you want to use note then try

```md
:::note
// YOUR NOTES HERE
:::
```

Try this for best practice

Comment (Ajay-Dhangar) [reply dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md:407]: and more info: https://codeharborhub.github.io/community/features/#admonitions

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by shreyash3087** [comment]
@Ajay-Dhangar Done

**Review by Ajay-Dhangar — APPROVED** [review]
OK

**Pull request merged**